test/core/end2end/tests/default_host.cc

```diff
@@ -216,10 +218,9 @@ static void test_invoke_simple_request(grpc_end2end_test_config config) {
 }
 
 void default_host(grpc_end2end_test_config config) {
-  if ((config.feature_mask & FEATURE_MASK_SUPPORTS_HOSTNAME_VERIFICATION) == 0)
-    return;
-  if ((config.feature_mask & FEATURE_MASK_SUPPORTS_DELAYED_CONNECTION) == 0)
+  if ((config.feature_mask & FEATURE_MASK_SUPPORTS_PER_CALL_CREDENTIALS) != 0) {
```

Comment:
note that this test wasn't being run for ANY test fixture (all of them support `FEATURE_MASK_SUPPORTS_HOSTNAME_VERIFICATION`). Right now, the only exception is h2_oauth and h2_ssl_proxy, as they require extra support that this test doesn't provide.

In any case, it's now run for all other test fixtures. In particular, it validates the right authority header behavior for direct channels when run under the h2_sockpair fixture, for example.